src/include/duckdb/planner/operator/logical_create_index.hpp

```diff
@@ -10,37 +10,60 @@
 
 #include "duckdb/parser/parsed_data/create_index_info.hpp"
 #include "duckdb/planner/logical_operator.hpp"
+#include "duckdb/function/table_function.hpp"
 
 namespace duckdb {
 
 class LogicalCreateIndex : public LogicalOperator {
 public:
-	LogicalCreateIndex(TableCatalogEntry &table, vector<column_t> column_ids,
-	                   vector<unique_ptr<Expression>> expressions, unique_ptr<CreateIndexInfo> info)
-	    : LogicalOperator(LogicalOperatorType::LOGICAL_CREATE_INDEX), table(table), column_ids(column_ids),
-	      info(std::move(info)) {
-		for (auto &expr : expressions) {
+	LogicalCreateIndex(idx_t table_index_p, TableCatalogEntry &table_p, vector<column_t> column_ids_p,
+	                   TableFunction function_p, unique_ptr<FunctionData> bind_data_p,
+	                   vector<unique_ptr<Expression>> expressions_p, unique_ptr<CreateIndexInfo> info_p,
+	                   vector<string> names_p, vector<LogicalType> returned_types_p)
+	    : LogicalOperator(LogicalOperatorType::LOGICAL_CREATE_INDEX), table_index(table_index_p), table(table_p),
+	      column_ids(column_ids_p), function(move(function_p)), bind_data(move(bind_data_p)), info(std::move(info_p)),
+	      names(names_p), returned_types(returned_types_p) {
+
+		for (auto &expr : expressions_p) {
 			this->unbound_expressions.push_back(expr->Copy());
 		}
-		this->expressions = move(expressions);
+		this->expressions = move(expressions_p);
+
+		if (column_ids.empty()) {
+			throw BinderException("CREATE INDEX does not refer to any columns in the base table!");
+		}
 	}
 
+	//! The table index in the current bind context
+	idx_t table_index;
 	//! The table to create the index for
 	TableCatalogEntry &table;
 	//! Column IDs needed for index creation
 	vector<column_t> column_ids;
+	//! The function that is called
+	TableFunction function;
+	//! The bind data of the function
+	unique_ptr<FunctionData> bind_data;
 	// Info for index creation
 	unique_ptr<CreateIndexInfo> info;
+	//! The names of the columns
+	vector<string> names;
+	//! The types of ALL columns that can be returned by the table scan
+	vector<LogicalType> returned_types;
+
 	//! Unbound expressions to be used in the optimizer
 	vector<unique_ptr<Expression>> unbound_expressions;
 
 public:
+	vector<ColumnBinding> GetColumnBindings() override;
```

Comment:
Why did you override `GetColumnBindings` and rework `ResolveTypes`? Can it not stay the way it was?

This operator seems similar to, e.g., `LogicalCopyToFile` now, which also does not override/rework these functions.

Reply:
New `CreateIndexInfo` to keep track of the types for the table scan. Because the type field of an operator keeps track of the types that it returns but this operator does not return any types.